graylog2-web-interface/src/components/indices/IndexSetFieldTypesList.tsx

```diff
@@ -0,0 +1,204 @@
+        setActivePage(newPage);
+      }
+
+      if (newPageSize) {
+        updateTableLayout({ perPage: newPageSize });
+      }
+    }, [updateTableLayout],
+  );
+
+  const onPageSizeChange = useCallback((newPageSize: number) => {
+    setActivePage(1);
+    updateTableLayout({ perPage: newPageSize });
+  }, [updateTableLayout]);
+
+  const onSortChange = useCallback((newSort: Sort) => {
+    setActivePage(1);
+    updateTableLayout({ sort: newSort });
+  }, [updateTableLayout]);
+
+  const onSearch = useCallback((newQuery: string) => {
+    setActivePage(1);
+    setQuery(newQuery);
+  }, []);
+
+  const onResetSearch = useCallback(() => onSearch(''), [onSearch]);
+
+  const onColumnsChange = useCallback((displayedAttributes: Array<string>) => {
+    updateTableLayout({ displayedAttributes });
+  }, [updateTableLayout]);
+  const { isLoading, data: { list, pagination, attributes }, refetch } = useIndexSetFieldTypes(indexSetId, searchParams, { enabled: !isLoadingLayoutPreferences });
+
+  const customColumnRenderers = useMemo(() => ({
+    attributes: {
+      type: {
+        renderCell: (item: string) => <span>{fieldTypes[item]}</span>,
+      },
+      is_custom: {
+        renderHeader: () => null,
+        renderCell: (isCustom: boolean) => (isCustom ? <Label bsStyle="primary">custom</Label> : null),
+      },
+    },
+  }), [fieldTypes]);
+
+  const openEditModal = useCallback((fieldType: IndexSetFieldType) => {
+    handleOnOpen(fieldType);
+  }, [handleOnOpen]);
+
+  const renderActions = useCallback((fieldType: IndexSetFieldType) => (
+    <>
+      <Button onClick={() => openEditModal(fieldType)}
+              role="button"
+              bsSize="xsmall"
+              disabled={fieldType.isReserved}
+              title={`${fieldType.isCustom ? 'Edit' : 'Add'} custom field type for ${fieldType.fieldName}`}
+              tabIndex={0}>
+        {fieldType.isCustom ? 'Edit' : 'Add'}
+      </Button>
+      {/* fieldType.isCustom && (
+      <Button onClick={onRemove}
+              role="button"
+              bsSize="xsmall"
+              bsStyle="danger"
+              title="Remove custom type"
+              tabIndex={0}>
+        Delete
+      </Button>
+      ) */}
+    </>
+  ), [openEditModal]);
+
+  if (isLoadingLayoutPreferences || isLoading) {
+    return <Spinner />;
+  }
+
+  return (
+    <>
+      <PaginatedList onChange={onPageChange}
+                     totalItems={pagination?.total}
+                     pageSize={layoutConfig.pageSize}
+                     activePage={activePage}
+                     showPageSizeSelect={false}
+                     useQueryParameter={false}>
+        <div style={{ marginBottom: '5px' }}>
+          <SearchForm focusAfterMount
+                      onSearch={onSearch}
+                      queryHelpComponent={<QueryHelper entityName="search" commonFields={['id', 'title']} />}
+                      topMargin={0}
+                      onReset={onResetSearch} />
```

Comment:
I was wondering why you just removed the input? My idea was more to fix the related logic, since it is a helpful functionality and usually we have a search inout for these tables.

Reply:
I removed it because filtering is not supported on BE for now